### ISSUE TITLE
PHPFMT not working, even if PHP location is set in preferences

### ISSUE THREAD

**Issue by David-van-den-Bor** [dericofilho/sublime-phpfmt#152]
Hi Derico :-)

I have a question regarding the PHPFMT plugin. I've set the location that the plugin has to check for my version of PHP (I use AMPPS instead of MAMP on Mac OSX)  My settings are:

```json
{
	"debug": true,
	"format_on_save": false,
	"php_bin": "/Applications/AMPPS/php-5.6/bin/php"
}
```

The PHP bin address points exactly to the folder containing the mac OSX PHP executable, but when selecting any of the PHPFMT tasks in Sublime (like "format" ect.), they do not fire(?) Any idea what could be the cause of this?

Kind regards and thanx in advance :-)
David van den Bor

**Comment by David-van-den-Bor** [comment]
Added below: this is the notification that appears in the console after firing "PHPFMT format" in Sublime text:

```c
cmd_fmt:  ['/Applications/AMPPS/php-5.6/bin/php', '/Users/davidvandenbor/Library/Application Support/Sublime Text 3/Packages/phpfmt/fmt.phar', '--config=/Users/davidvandenbor/Library/Application Support/Sublime Text 3/Packages/phpfmt/php.tools.ini', '--smart_linebreak_after_curly', '-v', '-o=-', '-']
```

**Comment by David-van-den-Bor** [comment]
Another issue: When selecting ```phpfmt: analyse this``` from the menu, Sublimes fires the dialog below

```console
Autocomplete and autoimport need building a database of terms. Please, fill in below with the proper location for this database. Keep in mind that the best location is always the root of the project, therefore limiting the size and ensuring speed.
```

Following this, the Sublime text debugging console spits out:

```console
phpfmt (oracle file): not found -- dialog
```

Everytime I choose the analyse command, I keep getiing the message to build the database, again and again and again and again :-)

Kind regards and thanx in advance :-)
David van den Bor

**Comment by dericofilho** [comment]
@David-van-den-Bor I am sorry for all these inconveniences. Please, do not mix up subjects into the same issue. Open separate issues so I can help you with all of them.

Regarding PHP binary address. Please, paste here the full console output (after `phpfmt: format now` is invoked) with the debug flag on. I really need to understand whether this is a permission issue. If it turns out to be a permission problem, then you will need to fix them on your computer accordingly.

Regarding `phpfmt: analyse this`, please, open a new issue. But probably, once your PHP configuration is done correctly, this will solve by itself.

**Comment by David-van-den-Bor** [comment]
Hi Carlos! Thank you for responding so quickly :-)

I pasted the output of the Sublime text console here. This is right after invoking "PHPFMT format" from Sublime:

```console
phpfmt (fmt.phar version) err:
 cmd_fmt:  ['/Applications/AMPPS/php-5.6/bin/php', '/Users/davidvandenbor/Library/Application Support/Sublime Text 3/Packages/phpfmt/fmt.8.9.0.phar', '--config=/Users/davidvandenbor/Library/Application Support/Sublime Text 3/Packages/phpfmt/php.tools.ini', '--smart_linebreak_after_curly', '-v', '-o=-', '-']
```

Maybe this gives some idea of what's going on. I added a screenshot below of the directory where PHPFMT looks for the PHP executable:

![schermafbeelding 2015-06-25 om 15 33 38](https://cloud.githubusercontent.com/assets/5694412/8355466/bb5aed1e-1b4f-11e5-817e-cb03806428e6.png)

**Comment by dericofilho** [comment]
@David-van-den-Bor hi,  this log output is incomplete for sure. There should be more lines among them a Python error complaining a missing executable.

**Comment by David-van-den-Bor** [comment]
Weird! The console really ends on these lines. I'll paste some of the lines that came **before** the final console statement. May be a clue there:

```console
phpfmt (php_ver) cmd:
 ['/Applications/AMPPS/php-5.6/bin/php', '-v']
phpfmt (php_ver) out:
 PHP 5.6.6 (cli) (built: Feb 24 2015 17:03:08) 
Copyright (c) 1997-2015 The PHP Group
Zend Engine v2.6.0, Copyright (c) 1998-2015 Zend Technologies

phpfmt (php_ver) err:
 
phpfmt (php version) out:
 PHP 5.6.6 (cli) (built: Feb 24 2015 17:03:08) 
Copyright (c) 1997-2015 The PHP Group
Zend Engine v2.6.0, Copyright (c) 1998-2015 Zend Technologies

phpfmt (php version) err:
 
phpfmt (fmt.phar version) out:
 /Users/davidvandenbor/Library/Application Support/Sublime Text 3/Packages/phpfmt/fmt.8.9.0.phar 8.9.0

phpfmt (fmt.phar version) err:
 
cmd_fmt:  ['/Applications/AMPPS/php-5.6/bin/php', '/Users/davidvandenbor/Library/Application Support/Sublime Text 3/Packages/phpfmt/fmt.8.9.0.phar', '--config=/Users/davidvandenbor/Library/Application Support/Sublime Text 3/Packages/phpfmt/php.tools.ini', '--smart_linebreak_after_curly', '-v', '-o=-', '-']
```

**Comment by dericofilho** [comment]
@David-van-den-Bor why are you using php55compat mode?

**Comment by dericofilho** [comment]
@David-van-den-Bor Please, manually update to https://github.com/dericofilho/sublime-phpfmt/commit/f1e0885f5ddc4886690b6dd154f247fb941460e7 and give all messages available, including Python error messages.

**Comment by David-van-den-Bor** [comment]
I manually replaced the PHPFMT.py with the one that you changed, opened up Sublime and ran the "PHPFMT format now" command again. I copied the complete output of the console below:

```console
startup, version: 3092 osx x64 channel: dev
executable: /Applications/Sublime Text.app/Contents/MacOS/Sublime Text
working dir: /
packages path: /Users/davidvandenbor/Library/Application Support/Sublime Text 3/Packages
state path: /Users/davidvandenbor/Library/Application Support/Sublime Text 3/Local
zip path: /Applications/Sublime Text.app/Contents/MacOS/Packages
zip path: /Users/davidvandenbor/Library/Application Support/Sublime Text 3/Installed Packages
ignored_packages: ["SHS-Bourbon-Neat", "SHS-Compass-Variables", "SHS-Bourbon", "Vintage", "SHS-Compass-Mixins", "Markdown"]
ignoring deprecated syntax definition: Packages/HTML5/HTML.tmLanguage
ignoring deprecated syntax definition: Packages/SublimeLinter/fixed-syntaxes/HTML/HTML.tmLanguage
ignoring deprecated syntax definition: Packages/SublimeLinter/fixed-syntaxes/Rails/HTML (Rails).tmLanguage
pre session restore time: 0.380018
using gpu buffer for window
first paint time: 0.461075
using gpu buffer for window
startup time: 0.56571
reloading plugin Default.block
reloading plugin Default.comment
reloading plugin Default.convert_syntax
reloading plugin Default.copy_path
reloading plugin Default.delete_word
reloading plugin Default.detect_indentation
reloading plugin Default.duplicate_line
reloading plugin Default.echo
reloading plugin Default.exec
reloading plugin Default.fold
reloading plugin Default.font
reloading plugin Default.goto_line
reloading plugin Default.history_list
reloading plugin Default.indentation
reloading plugin Default.kill_ring
reloading plugin Default.mark
reloading plugin Default.new_templates
reloading plugin Default.open_context_url
reloading plugin Default.open_file_settings
reloading plugin Default.open_in_browser
reloading plugin Default.pane
reloading plugin Default.paragraph
reloading plugin Default.paste_from_history
reloading plugin Default.quick_panel
reloading plugin Default.run_all_syntax_tests
reloading plugin Default.save_on_focus_lost
reloading plugin Default.scroll
reloading plugin Default.set_unsaved_view_name
reloading plugin Default.side_bar
reloading plugin Default.sort
reloading plugin Default.swap_line
reloading plugin Default.switch_file
reloading plugin Default.symbol
reloading plugin Default.transform
reloading plugin Default.transpose
reloading plugin Default.trim_trailing_white_space
reloading plugin CSS.css_completions
reloading plugin Diff.diff
reloading plugin HTML.encode_html_entities
reloading plugin HTML.html_completions
reloading plugin 0_package_control_loader.00-package_control
reloading plugin 0_package_control_loader.02-bz2
reloading plugin AlignTab.aligner
reloading plugin AlignTab.aligntab
reloading plugin AlignTab.hist
reloading plugin AlignTab.parser
reloading plugin AlignTab.table
reloading plugin AlignTab.wclen
reloading plugin AppendSemiColon.AppendSemiColon
reloading plugin Bower.Bower
reloading plugin Bracketeer.bracketeer
reloading plugin BracketGuard.BracketGuard
reloading plugin BracketHighlighter.bh_core
reloading plugin BracketHighlighter.bh_logging
reloading plugin BracketHighlighter.bh_plugin
reloading plugin BracketHighlighter.bh_regions
reloading plugin BracketHighlighter.bh_remove
reloading plugin BracketHighlighter.bh_rules
reloading plugin BracketHighlighter.bh_search
reloading plugin BracketHighlighter.bh_swapping
reloading plugin BracketHighlighter.bh_wrapping
reloading plugin BracketHighlighter.ure
reloading plugin Browser Refresh.BrowserRefresh
reloading plugin Color Highlighter.ColorHighlighter
reloading plugin Color Highlighter.colors
reloading plugin CSS Extended Completions.cache
reloading plugin CSS Extended Completions.commands
reloading plugin CSS Extended Completions.completions
reloading plugin CSS Extended Completions.css_style_completions
reloading plugin CSS Extended Completions.extended_css_completions
reloading plugin CSS Extended Completions.location
reloading plugin CSS Extended Completions.project
reloading plugin CSS Extended Completions.settings
reloading plugin CSS Extended Completions.style_parser
reloading plugin CSS Format.css_format
reloading plugin CSS3.css3_completions
reloading plugin CSS3.rgb_to_hex
reloading plugin CSS3.validator
reloading plugin Deselect.Deselect
reloading plugin Emmet.emmet-plugin
reloading plugin ExpandRegion.__init__
reloading plugin ExpandRegion.expand_region_handler
reloading plugin ExpandRegion.expand_to_line
reloading plugin ExpandRegion.expand_to_quotes
reloading plugin ExpandRegion.expand_to_regex_set
reloading plugin ExpandRegion.expand_to_semantic_unit
reloading plugin ExpandRegion.expand_to_symbols
reloading plugin ExpandRegion.expand_to_word
reloading plugin ExpandRegion.expand_to_word_with_dots
reloading plugin ExpandRegion.expand_to_xml_node
reloading plugin ExpandRegion.ExpandRegion
reloading plugin ExpandRegion.html
reloading plugin ExpandRegion.javascript
reloading plugin ExpandRegion.utils
reloading plugin Gist.gist
reloading plugin Gist.helpers
reloading plugin Gist.request
reloading plugin Gist.settings
reloading plugin Goto-CSS-Declaration.goto_css_declaration
reloading plugin HTML5.encode_html_entities
reloading plugin HTML5.html_completions
reloading plugin Insert Filepath.InsertFilepath
reloading plugin Keymap Redefiner.Keymap Redefiner
reloading plugin LastModifiedIndicator.LastModifiedIndicator
reloading plugin LiveStyle.livestyle
reloading plugin MarkdownEditing.bootstrap
reloading plugin MarkdownEditing.custom_find_under_expand
reloading plugin MarkdownEditing.decide_title
reloading plugin MarkdownEditing.distraction_free_mode
reloading plugin MarkdownEditing.footnotes
reloading plugin MarkdownEditing.gather_missing_links
reloading plugin MarkdownEditing.indent_list_item
reloading plugin MarkdownEditing.indent_list_multiitem
reloading plugin MarkdownEditing.insert_references
reloading plugin MarkdownEditing.jumpToMarker
reloading plugin MarkdownEditing.lint
reloading plugin MarkdownEditing.list_markdown_references
reloading plugin MarkdownEditing.numbered_list
reloading plugin MarkdownEditing.paste_as_link
reloading plugin MarkdownEditing.paste_as_reference
reloading plugin MarkdownEditing.quote_indenting
reloading plugin MarkdownEditing.switch_list_bullet_type
reloading plugin MarkdownEditing.underlined_headers
reloading plugin Marked App Menu.marked
reloading plugin Minifier.Minify
reloading plugin Package Control.bootstrap
reloading plugin Package Control.Package Control
reloading plugin Pretty JSON.PrettyJson
[Errno 2] No such file or directory: 'jq'
reloading plugin RoboType.RoboTypePlugin
reloading plugin SassBeautify.SassBeautify
reloading plugin SCSS.scss_completions
reloading plugin SideBarEnhancements.SideBar
reloading plugin SideBarEnhancements.SideBarAPI
reloading plugin SideBarEnhancements.SideBarDefaultDisable
reloading plugin SnippetMaker.snippet_maker
reloading plugin SublimeLinter-contrib-scss-lint.linter
SublimeLinter: scss linter loaded 
reloading plugin SublimeLinter-csslint.linter
SublimeLinter: csslint linter loaded 
reloading plugin SublimeLinter-html-tidy.linter
SublimeLinter: htmltidy linter loaded 
reloading plugin SublimeLinter-jshint.linter
SublimeLinter: jshint linter loaded 
reloading plugin SublimeLinter-php.linter
SublimeLinter: php linter loaded 
reloading plugin SublimeLinter-phplint.linter
SublimeLinter: phplint linter loaded 
reloading plugin SuperNavigator.SuperNavigator
reloading plugin Tag.Edit
reloading plugin Tag.Tag
reloading plugin Tag.tag_classes
reloading plugin Tag.tag_close_tag
reloading plugin Tag.tag_close_tag_on_slash
reloading plugin Tag.tag_indent
reloading plugin Tag.tag_insert_as_tag
reloading plugin Tag.tag_lint
reloading plugin Tag.tag_remove
reloading plugin Tag.tag_remove_attributes
reloading plugin Text Pastry.text_pastry
reloading plugin Text Pastry.text_pastry_addons
reloading plugin Text Pastry.text_pastry_selection
reloading plugin ToggleExclude.ToggleExclude
reloading plugin Trimmer.Trimmer
reloading plugin WordPress Developer Resources.SublimeWPDeveloperResources
reloading plugin FixMyJS.FixMyJS
reloading plugin FixMyJS.node_bridge
reloading plugin Gulp.base_command
reloading plugin Gulp.gulp
reloading plugin Gulp.progress_notifier
reloading plugin HTML-CSS-JS Prettify.HTMLPrettify
reloading plugin MacTerminal.open_mac_terminal
reloading plugin phpfmt.phpfmt
reloading plugin StringUtilities.stringutilities
reloading plugin SublimeCodeIntel.ordereddict
reloading plugin SublimeCodeIntel.SublimeCodeIntel
reloading Packages/User/phpfmt.sublime-settings
ignoring deprecated syntax definition: Packages/HTML5/HTML.tmLanguage
ignoring deprecated syntax definition: Packages/SublimeLinter/fixed-syntaxes/HTML/HTML.tmLanguage
ignoring deprecated syntax definition: Packages/SublimeLinter/fixed-syntaxes/Rails/HTML (Rails).tmLanguage
reloading plugin SublimeLinter.commands
reloading plugin SublimeLinter.sublimelinter
reloading plugin Takana.takana
***************************************
Takana plugin v0.4
Takana: connecting
Takana: connection failed
plugins loaded
StringUtilities: Plugin Initialized
SublimeLinter: debug mode: off 
SublimeLinter: php activated: /usr/bin/php 
SublimeLinter: WARNING: phplint deactivated, cannot locate 'phplint' 
Emmet: No need to update PyV8
reloading /Users/davidvandenbor/Library/Application Support/Sublime Text 3/Packages/User/phpfmt.sublime-settings
reloading /Users/davidvandenbor/Library/Application Support/Sublime Text 3/Packages/User/Default (OSX).sublime-keymap
Package Control: Skipping automatic upgrade, last run at 2015-06-25 16:52:48, next run at 2015-06-25 17:52:48 or after
Selfupdate
WINDOW COMMAND ENABLED True
WINDOW COMMAND ENABLED True
WINDOW COMMAND ENABLED True
WINDOW COMMAND ENABLED True
WINDOW COMMAND ENABLED True
WINDOW COMMAND ENABLED False
phpfmt (php_ver) cmd:
 ['/Applications/AMPPS/php-5.6/bin/php', '-v']
phpfmt (php_ver) out:
 PHP 5.6.6 (cli) (built: Feb 24 2015 17:03:08) 
Copyright (c) 1997-2015 The PHP Group
Zend Engine v2.6.0, Copyright (c) 1998-2015 Zend Technologies

phpfmt (php_ver) err:
 
phpfmt (php version) out:
 PHP 5.6.6 (cli) (built: Feb 24 2015 17:03:08) 
Copyright (c) 1997-2015 The PHP Group
Zend Engine v2.6.0, Copyright (c) 1998-2015 Zend Technologies

phpfmt (php version) err:
 
phpfmt (fmt.phar version) out:
 /Users/davidvandenbor/Library/Application Support/Sublime Text 3/Packages/phpfmt/fmt.phar 9.2.1

phpfmt (fmt.phar version) err:
 
cmd_fmt:  ['/Applications/AMPPS/php-5.6/bin/php', '/Users/davidvandenbor/Library/Application Support/Sublime Text 3/Packages/phpfmt/fmt.phar', '--config=/Users/davidvandenbor/Library/Application Support/Sublime Text 3/Packages/phpfmt/php.tools.ini', '--smart_linebreak_after_curly', '-v', '-o=-', '-']
p:
 255
err:
 Used passes: RTrim, EliminateDuplicatedEmptyLines, LeftAlignComment, Reindent, ReindentAndAlignObjOps, ReindentIfColonBlocks, ReindentLoopColonBlocks, ReindentColonBlocks, SplitCurlyCloseAndTokens, ResizeSpaces, StripExtraCommaInList, MergeDoubleArrowAndArray, MergeCurlyCloseAndDoWhile, MergeParenCloseWithCurlyOpen, NormalizeLnAndLtrimLines, ExtraCommaInArray, SmartLnAfterCurlyOpen, AddMissingCurlyBraces, OrderUseClauses, NormalizeIsNotEquals, RemoveIncludeParentheses, TwoCommandsInSameLine
PHP Fatal error:  Call to undefined function token_get_all() in phar:///Users/davidvandenbor/Library/Application Support/Sublime Text 3/Packages/phpfmt/fmt.phar/fmt.stub.php on line 3619
```

**Comment by David-van-den-Bor** [comment]
With the above output, these were my PHPFMT settings:

```json
{
	"debug": true,
	"format_on_save": false,
	"passes":
	[
	],
	"php_bin": "/Applications/AMPPS/php-5.6/bin/php",
	"version": 2
}
```

**Comment by dericofilho** [comment]
@David-van-den-Bor your PHP installation is missing the tokenizer module. http://php.net/manual/en/tokenizer.installation.php 

Please, enable it and it will work just fine.

**Comment by David-van-den-Bor** [comment]
![schermafbeelding 2015-06-25 om 17 40 19](https://cloud.githubusercontent.com/assets/5694412/8358768/74b5612e-1b62-11e5-904a-4cd4ab008108.png)


Activated tokenizer module in AMPPS and confirming that the **plugin is working** now :-)
Regards, and thanxx for helping out!
